portfolio/src/main/webapp/script.js

```diff
@@ -33,23 +68,35 @@ function gotoRandomSite() {
   window.location.href = site;
 }
 
+/**
+ * Gets comments from datastore and stores them in Comment objects.
+ * Also updates the DOM with new comments.
+ */
 function getComments() {
   // Perform the fetch and store as promise
   var commentsPromise = fetch('/data');
   commentsPromise.then(response => response.json())
     .then((resJson) => {
-      comments = resJson;
```

Comment:
Do you need to reset commentData to empty array before populating it with response contents?